tests/cases/conformance/jsdoc/checkJsdocTypeTag6.ts

```diff
@@ -21,3 +21,26 @@ function add2(a, b) { return a + b; }
 // TODO: Should be an error since signature doesn't match.
 /** @type {(a: number, b: number, c: number) => number} */
 function add3(a, b) { return a + b; }
+
+// Confirm initializers are compatible.
+// They can't have more parameters than the target.
+
+/** @type {() => void} */
+function func(more) {}
+
+/** @type {() => void} */
+const variable = function (more) {};
+
+/** @type {() => void} */
+const arrow = (more) => {};
+
+({
+  /** @type {() => void} */
+  method(more) {},
```

Comment:
I don't understand why this is an error but `function add3` isn't an error above. Is it because of the exemption of object literal methods?

Reply:
You can assign types with fewer parameters to types with more, just not the other way around.

That makes sense, I think: The target could be something general purpose and an implementation might only use some parameters, like maybe:
```ts
/**
 * @callback Plugin
 * @param {Node} tree
 * @param {unknown} options
 * @returns {void}
 */

/** @type {Plugin} */
function visit(tree) {}

Reply:
It's OK, although this isn't really assignment to the `Plugin` type either, it's supposed to act like type annotations on the parameters themselves. Like I said, it's a *little* odd that the parameters from the `@type` replace the ones from the function declaration, but it's probably OK.